Maven Daemon doesn't download Core Extensions from pluginRepository defined in settings.xml

On a project that I'm working on, we need to use a patched version of a Maven Extension (specified with `.mvn/extensions.xml`) that is stored in an internal Maven repository. When using Maven we are able to download the Maven Extension from a pluginRepository specified in the settings.xml (we use --settings option to point to a project specific settings.xml). When using mvnd this doesn't work.
From my investigation it seams that Maven Daemon doesn't resolve the settings.xml passed with `--settings` option when downloading Maven Core Extensions. In fact it tries to download it from `central (https://repo.maven.apache.org/maven2)`. In our settings.xml we use a mirror of Maven central for both dependencies and plugins. This makes me think that mvnd doesn't use the specified settings.xml at early stage when it needs to resolve Maven Core Extensions defined in `.mvn/extensions.xml`.

We are using mvnd version 1.0.2.

Partial stacktrace:
```
java.lang.RuntimeException: Could not initialize org.mvndaemon.mvnd.daemon.Server
	at org.mvndaemon.mvnd.daemon.Server.<init>(Server.java:171)
	at org.mvndaemon.mvnd.daemon.Server.main(Server.java:103)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchStandard(Launcher.java:292)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:208)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:361)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:314)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:74)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.mvndaemon.mvnd.daemon.Server.<init>(Server.java:136)
	... 7 more
Caused by: org.apache.maven.cli.internal.ExtensionResolutionException: Extension <omitted> or one of its dependencies could not be resolved: Plugin <omitted> or one of its dependencies could not be resolved:
	Could not find artifact <omitted> in central (https://repo.maven.apache.org/maven2)

	at org.apache.maven.cli.internal.BootstrapCoreExtensionManager.resolveExtension(BootstrapCoreExtensionManager.java:182)
	at org.apache.maven.cli.internal.BootstrapCoreExtensionManager.resolveCoreExtensions(BootstrapCoreExtensionManager.java:115)
	at org.apache.maven.cli.internal.BootstrapCoreExtensionManager.loadCoreExtensions(BootstrapCoreExtensionManager.java:99)
	at org.apache.maven.cli.DaemonMavenCli.loadCoreExtensions(DaemonMavenCli.java:765)
	at org.apache.maven.cli.DaemonMavenCli.container(DaemonMavenCli.java:668)
	at org.apache.maven.cli.DaemonMavenCli.<init>(DaemonMavenCli.java:205)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	... 10 more
Caused by: org.apache.maven.plugin.PluginResolutionException: Plugin <omitted> or one of its dependencies could not be resolved:
	Could not find artifact <omitted> in central (https://repo.maven.apache.org/maven2)

	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveInternal(DefaultPluginDependenciesResolver.java:234)
	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolveCoreExtension(DefaultPluginDependenciesResolver.java:159)
	at org.apache.maven.cli.internal.BootstrapCoreExtensionManager.resolveExtension(BootstrapCoreExtensionManager.java:174)
	... 16 more
----- End of the daemon output -----

	at org.mvndaemon.mvnd.client.DaemonConnector.startDaemon(DaemonConnector.java:330)
	at org.mvndaemon.mvnd.client.DaemonConnector.connect(DaemonConnector.java:123)
	at org.mvndaemon.mvnd.client.DefaultClient.execute(DefaultClient.java:353)
	at org.mvndaemon.mvnd.client.DefaultClient.main(DefaultClient.java:171)
	at java.base@22.0.2/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
```

--- a/daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java
+++ b/daemon/src/main/java/org/apache/maven/cli/DaemonMavenCli.java
@@ -156,8 +156,6 @@ public class DaemonMavenCli implements DaemonCli {
     private static final String UNABLE_TO_FIND_ROOT_PROJECT_MESSAGE = "Unable to find the root directory. Create a "
             + DOT_MVN + " directory in the project root directory to identify it.";
 
-    private static final String EXTENSIONS_FILENAME = DOT_MVN + "/extensions.xml";
-
     private static final String MVN_MAVEN_CONFIG = DOT_MVN + "/maven.config";
 
     public static final String STYLE_COLOR_PROPERTY = "style.color";
@@ -174,23 +172,16 @@ public class DaemonMavenCli implements DaemonCli {
 
     private final ClassWorld classWorld;
 
-    private final DefaultPlexusContainer container;
-
-    private final EventSpyDispatcher eventSpyDispatcher;
-
-    private final ModelProcessor modelProcessor;
-
-    private final Maven maven;
-
-    private final MavenExecutionRequestPopulator executionRequestPopulator;
-
-    private final ToolchainsBuilder toolchainsBuilder;
-
-    private final DefaultSecDispatcher dispatcher;
-
-    private final Map<String, ConfigurationProcessor> configurationProcessors;
-
-    private final LoggingExecutionListener executionListener;
+    // lazily created
+    private DefaultPlexusContainer container;
+    private EventSpyDispatcher eventSpyDispatcher;
+    private ModelProcessor modelProcessor;
+    private Maven maven;
+    private MavenExecutionRequestPopulator executionRequestPopulator;
+    private ToolchainsBuilder toolchainsBuilder;
+    private DefaultSecDispatcher dispatcher;
+    private Map<String, ConfigurationProcessor> configurationProcessors;
+    private LoggingExecutionListener executionListener;
 
     /** Non-volatile, assuming that it is accessed only from the main thread */
     private BuildEventListener buildEventListener = BuildEventListener.dummy();
@@ -201,17 +192,6 @@ public class DaemonMavenCli implements DaemonCli {
         plexusLoggerManager = new Slf4jLoggerManager();
 
         this.classWorld = ((ClassRealm) Thread.currentThread().getContextClassLoader()).getWorld();
-
-        container = container();
-
-        eventSpyDispatcher = container.lookup(EventSpyDispatcher.class);
-        maven = container.lookup(Maven.class);
-        executionRequestPopulator = container.lookup(MavenExecutionRequestPopulator.class);
-        modelProcessor = createModelProcessor(container);
-        configurationProcessors = container.lookupMap(ConfigurationProcessor.class);
-        toolchainsBuilder = container.lookup(ToolchainsBuilder.class);
-        dispatcher = (DefaultSecDispatcher) container.lookup(SecDispatcher.class, "maven");
-        executionListener = container.lookup(LoggingExecutionListener.class);
     }
 
     public int main(
@@ -237,7 +217,9 @@ public class DaemonMavenCli implements DaemonCli {
         Properties props = (Properties) System.getProperties().clone();
         ClassLoader tccl = Thread.currentThread().getContextClassLoader();
         try {
-            Thread.currentThread().setContextClassLoader(container.getContainerRealm());
+            if (container != null) {
+                Thread.currentThread().setContextClassLoader(container.getContainerRealm());
+            }
             initialize(cliRequest);
             environment(cliRequest.workingDirectory, clientEnv);
             cli(cliRequest);
@@ -245,7 +227,7 @@ public class DaemonMavenCli implements DaemonCli {
             logging(cliRequest);
             informativeCommands(cliRequest);
             version(cliRequest);
-            container(cliRequest);
+            container(cliRequest); // we have here all
             configure(cliRequest, eventSpyDispatcher, configurationProcessors);
             toolchains(cliRequest);
             populateRequest(cliRequest);
@@ -257,7 +239,9 @@ public class DaemonMavenCli implements DaemonCli {
             // pure user error, suppress stack trace
             return 1;
         } finally {
-            eventSpyDispatcher.close();
+            if (eventSpyDispatcher != null) {
+                eventSpyDispatcher.close();
+            }
             System.setProperties(props);
             Thread.currentThread().setContextClassLoader(tccl);
         }
@@ -628,17 +612,30 @@ public class DaemonMavenCli implements DaemonCli {
         return interpolator;
     }
 
-    void container(CliRequest cliRequest) {
+    void container(CliRequest cliRequest) throws Exception {
+        if (this.container == null) {
+            this.container = doCreateContainer(cliRequest);
+
+            this.eventSpyDispatcher = container.lookup(EventSpyDispatcher.class);
+            this.maven = container.lookup(Maven.class);
+            this.executionRequestPopulator = container.lookup(MavenExecutionRequestPopulator.class);
+            this.modelProcessor = createModelProcessor(container);
+            this.configurationProcessors = container.lookupMap(ConfigurationProcessor.class);
+            this.toolchainsBuilder = container.lookup(ToolchainsBuilder.class);
+            this.dispatcher = (DefaultSecDispatcher) container.lookup(SecDispatcher.class, "maven");
+            this.executionListener = container.lookup(LoggingExecutionListener.class);
+        }
+
         Map<String, Object> data = new HashMap<>();
         data.put("plexus", container);
         data.put("workingDirectory", cliRequest.workingDirectory);
         data.put("systemProperties", cliRequest.systemProperties);
         data.put("userProperties", cliRequest.userProperties);
         data.put("versionProperties", CLIReportingUtils.getBuildProperties());
-        eventSpyDispatcher.init(() -> data);
+        this.eventSpyDispatcher.init(() -> data);
     }
 
-    DefaultPlexusContainer container() throws Exception {
+    DefaultPlexusContainer doCreateContainer(CliRequest cliRequest) throws Exception {
         ClassRealm coreRealm = classWorld.getClassRealm("plexus.core");
         if (coreRealm == null) {
             coreRealm = classWorld.getRealms().iterator().next();
@@ -665,7 +662,7 @@ public class DaemonMavenCli implements DaemonCli {
                 })
                 .collect(Collectors.toList());
         List<CoreExtensionEntry> extensionsEntries =
-                loadCoreExtensions(extensions, coreRealm, coreEntry.getExportedArtifacts());
+                loadCoreExtensions(cliRequest, extensions, coreRealm, coreEntry.getExportedArtifacts());
         ClassRealm containerRealm = setupContainerRealm(classWorld, coreRealm, extClassPath, extensionsEntries);
 
         ContainerConfiguration cc = new DefaultContainerConfiguration()
@@ -717,7 +714,11 @@ public class DaemonMavenCli implements DaemonCli {
     }
 
     private List<CoreExtensionEntry> loadCoreExtensions(
-            List<CoreExtension> extensions, ClassRealm containerRealm, Set<String> providedArtifacts) throws Exception {
+            CliRequest cliRequest,
+            List<CoreExtension> extensions,
+            ClassRealm containerRealm,
+            Set<String> providedArtifacts)
+            throws Exception {
         if (extensions.isEmpty()) {
             return Collections.emptyList();
         }
@@ -737,8 +738,6 @@ public class DaemonMavenCli implements DaemonCli {
         });
         MavenExecutionRequestPopulator executionRequestPopulator = null;
         try {
-            CliRequest cliRequest = new CliRequest(new String[0], classWorld);
-            cliRequest.commandLine = new CommandLine.Builder().build();
             container.setLookupRealm(null);
             container.setLoggerManager(plexusLoggerManager);
             container.getLoggerManager().setThresholds(cliRequest.request.getLoggingLevel());

